addons/mail/models/res_partner.py

```diff
@@ -41,6 +43,13 @@ def _compute_im_status(self):
         if odoobot in self:
             odoobot.im_status = 'bot'
 
+    def _compute_custom_status(self):
+        for partner in self:
+            users = partner.with_context(active_test=False).user_ids
+            internal_users = users - users.filtered('share')
+            main_user = internal_users[0] if len(internal_users) > 0 else users[0] if len(users) > 0 else self.env['res.users']
```

Comment:
I feel it's more idiomatic/easier to read.
```suggestion
            main_user = next((u for u in partner.with_context(active_test=False).user_ids if not u.share), self.env["res.users"])
```

Reply:
:+1: 
Just FYI the same logic is used in `_to_store` so it might be worth to create a method to make it more DRY

Reply:
If we are using this in several places we could even have a computed field :)

Reply:
That's what I was thinking, probably for another PR (?)

Reply:
The compute probably but the way to retrieve the main user here can already be simplified. It's much more pythonic imo.